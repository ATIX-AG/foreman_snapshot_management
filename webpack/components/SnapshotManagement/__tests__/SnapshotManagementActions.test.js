import { testActionSnapshotWithFixtures } from 'react-redux-test-utils';
import { API } from 'foremanReact/redux/API';

import {
  loadSnapshotList,
  snapshotDeleteAction,
  snapshotRollbackAction,
  snapshotUpdateAction,
} from '../SnapshotManagementActions';

jest.mock('foremanReact/redux/API/API');

const successResponse = {
  data: 'some-data',
};

const doLoadSnapshotList = (hostId, serverMock) => {
  API.get.mockImplementation(serverMock);

  return loadSnapshotList(hostId);
};

const doDeleteSnapshot = (host, rowData, serverMock) => {
  API.delete.mockImplementation(serverMock);

  return snapshotDeleteAction(host, rowData);
};

const doRollbackSnapshot = (host, rowData, serverMock) => {
  API.put.mockImplementation(serverMock);

  return snapshotRollbackAction(host, rowData);
};

const doUpdateSnapshot = (host, rowData, serverMock) => {
  API.put.mockImplementation(serverMock);

  return snapshotUpdateAction(host, rowData);
};

const listFixtures = {
  'should load snapshots and success': () =>
    doLoadSnapshotList(42, async () => successResponse),

  'should load snapshots and fail': () =>
    doLoadSnapshotList(42, async () => {
      throw new Error('some-error');
    }),
};

describe('Snapshot list actions', () =>
  testActionSnapshotWithFixtures(listFixtures));

const deleteFixtures = {
  'should delete snapshot and success': () =>
    doDeleteSnapshot(
      { id: 42, name: 'deep.thought' },
      { id: 'snapshot-0815', name: 'Savegame' },
      async () => successResponse
    ),

  'should load snapshots and fail': () =>
    doDeleteSnapshot(
      { id: 42, name: 'deep.thought' },
      { id: 'snapshot-0815', name: 'Savegame' },
      async () => {
        throw new Error('some-error');
      }
    ),
};

describe('Snapshot snapshot-delete actions', () =>
  testActionSnapshotWithFixtures(deleteFixtures));

const rollbackFixtures = {
  'should rollback snapshot and success': () =>
    doRollbackSnapshot(
      { id: 42, name: 'deep.thought' },
      { id: 'snapshot-0815', name: 'Savegame' },
      async () => successResponse
    ),

  'should load snapshots and fail': () =>
    doRollbackSnapshot(
      { id: 42, name: 'deep.thought' },
      { id: 'snapshot-0815', name: 'Savegame' },
      async () => {
        throw new Error('some-error');
      }
    ),
};

describe('Snapshot snapshot-rollback actions', () =>
  testActionSnapshotWithFixtures(rollbackFixtures));

const updateFixtures = {
  'should update snapshot and success': () =>
    doUpdateSnapshot(
      { id: 42, name: 'deep.thought' },
      {
        id: 'snapshot-0815',
        name: 'Savegame',
        description: 'Saw the three headed monkey!',
      },
      async () => successResponse
    ),

  'should load snapshots and fail': () =>
    doUpdateSnapshot(
      { id: 42, name: 'deep.thought' },
      {
        id: 'snapshot-0815',
        name: 'Savegame',
        description: 'Saw the three headed monkey!',
      },
      async () => {
        throw new Error('some-error');
      }
    ),
};

describe('Snapshot snapshot-update actions', () =>
  testActionSnapshotWithFixtures(updateFixtures));
