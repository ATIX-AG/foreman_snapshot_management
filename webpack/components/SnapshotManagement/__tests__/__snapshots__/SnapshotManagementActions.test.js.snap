// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot list actions should load snapshots and fail 1`] = `
Array [
  Array [
    Object {
      "payload": Object {
        "hostId": 42,
      },
      "type": "FOREMAN_SNAPSHOT_MANAGEMENT_SNAPSHOT_LIST_REQUEST",
    },
  ],
  Array [
    Object {
      "payload": Object {
        "hostId": 42,
      },
      "response": [Error: some-error],
      "type": "FOREMAN_SNAPSHOT_MANAGEMENT_SNAPSHOT_LIST_FAILURE",
    },
  ],
]
`;

exports[`Snapshot list actions should load snapshots and success 1`] = `
Array [
  Array [
    Object {
      "payload": Object {
        "hostId": 42,
      },
      "type": "FOREMAN_SNAPSHOT_MANAGEMENT_SNAPSHOT_LIST_REQUEST",
    },
  ],
  Array [
    Object {
      "payload": Object {
        "hostId": 42,
      },
      "response": "some-data",
      "type": "FOREMAN_SNAPSHOT_MANAGEMENT_SNAPSHOT_LIST_SUCCESS",
    },
  ],
]
`;

exports[`Snapshot snapshot-delete actions should delete snapshot and success 1`] = `
Array [
  Array [
    Object {
      "payload": Object {
        "host": Object {
          "id": 42,
          "name": "deep.thought",
        },
        "id": "snapshot-0815",
      },
      "type": "FOREMAN_SNAPSHOT_MANAGEMENT_SNAPSHOT_DELETE_REQUEST",
    },
  ],
  Array [
    Object {
      "payload": Object {
        "key": "FOREMAN_SNAPSHOT_MANAGEMENT_SNAPSHOT_DELETE_SUCCESS",
        "toast": Object {
          "key": "FOREMAN_SNAPSHOT_MANAGEMENT_SNAPSHOT_DELETE_SUCCESS",
          "message": "Successfully removed Snapshot \\"Savegame\\" from host deep.thought",
          "type": "success",
        },
      },
      "type": "toasts/addToast",
    },
  ],
  Array [
    [Function],
  ],
  Array [
    Object {
      "payload": Object {
        "host": Object {
          "id": 42,
          "name": "deep.thought",
        },
        "id": "snapshot-0815",
      },
      "response": "some-data",
      "type": "FOREMAN_SNAPSHOT_MANAGEMENT_SNAPSHOT_DELETE_SUCCESS",
    },
  ],
]
`;

exports[`Snapshot snapshot-delete actions should load snapshots and fail 1`] = `
Array [
  Array [
    Object {
      "payload": Object {
        "host": Object {
          "id": 42,
          "name": "deep.thought",
        },
        "id": "snapshot-0815",
      },
      "type": "FOREMAN_SNAPSHOT_MANAGEMENT_SNAPSHOT_DELETE_REQUEST",
    },
  ],
  Array [
    Object {
      "payload": Object {
        "key": "FOREMAN_SNAPSHOT_MANAGEMENT_SNAPSHOT_DELETE_FAILURE",
        "toast": Object {
          "key": "FOREMAN_SNAPSHOT_MANAGEMENT_SNAPSHOT_DELETE_FAILURE",
          "message": "Error occurred while removing Snapshot: Error: some-error",
          "type": "error",
        },
      },
      "type": "toasts/addToast",
    },
  ],
  Array [
    Object {
      "payload": Object {
        "host": Object {
          "id": 42,
          "name": "deep.thought",
        },
        "id": "snapshot-0815",
      },
      "response": [Error: some-error],
      "type": "FOREMAN_SNAPSHOT_MANAGEMENT_SNAPSHOT_DELETE_FAILURE",
    },
  ],
]
`;

exports[`Snapshot snapshot-rollback actions should load snapshots and fail 1`] = `
Array [
  Array [
    Object {
      "payload": Object {
        "host": Object {
          "id": 42,
          "name": "deep.thought",
        },
        "id": "snapshot-0815",
      },
      "type": "FOREMAN_SNAPSHOT_MANAGEMENT_SNAPSHOT_ROLLBACK_REQUEST",
    },
  ],
  Array [
    Object {
      "payload": Object {
        "key": "FOREMAN_SNAPSHOT_MANAGEMENT_SNAPSHOT_ROLLBACK_FAILURE",
        "toast": Object {
          "key": "FOREMAN_SNAPSHOT_MANAGEMENT_SNAPSHOT_ROLLBACK_FAILURE",
          "message": "Error occurred while rolling back VM: Error: some-error",
          "type": "error",
        },
      },
      "type": "toasts/addToast",
    },
  ],
  Array [
    Object {
      "payload": Object {
        "host": Object {
          "id": 42,
          "name": "deep.thought",
        },
        "id": "snapshot-0815",
      },
      "response": [Error: some-error],
      "type": "FOREMAN_SNAPSHOT_MANAGEMENT_SNAPSHOT_ROLLBACK_FAILURE",
    },
  ],
]
`;

exports[`Snapshot snapshot-rollback actions should rollback snapshot and success 1`] = `
Array [
  Array [
    Object {
      "payload": Object {
        "host": Object {
          "id": 42,
          "name": "deep.thought",
        },
        "id": "snapshot-0815",
      },
      "type": "FOREMAN_SNAPSHOT_MANAGEMENT_SNAPSHOT_ROLLBACK_REQUEST",
    },
  ],
  Array [
    Object {
      "payload": Object {
        "key": "FOREMAN_SNAPSHOT_MANAGEMENT_SNAPSHOT_ROLLBACK_SUCCESS",
        "toast": Object {
          "key": "FOREMAN_SNAPSHOT_MANAGEMENT_SNAPSHOT_ROLLBACK_SUCCESS",
          "message": "Successfully rolled back Snapshot \\"Savegame\\" on host deep.thought",
          "type": "success",
        },
      },
      "type": "toasts/addToast",
    },
  ],
  Array [
    Object {
      "payload": Object {
        "host": Object {
          "id": 42,
          "name": "deep.thought",
        },
        "id": "snapshot-0815",
      },
      "response": "some-data",
      "type": "FOREMAN_SNAPSHOT_MANAGEMENT_SNAPSHOT_ROLLBACK_SUCCESS",
    },
  ],
]
`;

exports[`Snapshot snapshot-update actions should load snapshots and fail 1`] = `
Array [
  Array [
    Object {
      "payload": Object {
        "host": Object {
          "id": 42,
          "name": "deep.thought",
        },
        "id": "snapshot-0815",
        "snapshot": Object {
          "description": "Saw the three headed monkey!",
          "name": "Savegame",
        },
      },
      "type": "FOREMAN_SNAPSHOT_MANAGEMENT_SNAPSHOT_UPDATE_REQUEST",
    },
  ],
  Array [
    Object {
      "payload": Object {
        "key": "FOREMAN_SNAPSHOT_MANAGEMENT_SNAPSHOT_UPDATE_FAILURE",
        "toast": Object {
          "key": "FOREMAN_SNAPSHOT_MANAGEMENT_SNAPSHOT_UPDATE_FAILURE",
          "message": "Error occurred while updating Snapshot: Error: some-error",
          "type": "error",
        },
      },
      "type": "toasts/addToast",
    },
  ],
  Array [
    Object {
      "payload": Object {
        "host": Object {
          "id": 42,
          "name": "deep.thought",
        },
        "id": "snapshot-0815",
      },
      "response": [Error: some-error],
      "type": "FOREMAN_SNAPSHOT_MANAGEMENT_SNAPSHOT_UPDATE_FAILURE",
    },
  ],
]
`;

exports[`Snapshot snapshot-update actions should update snapshot and success 1`] = `
Array [
  Array [
    Object {
      "payload": Object {
        "host": Object {
          "id": 42,
          "name": "deep.thought",
        },
        "id": "snapshot-0815",
        "snapshot": Object {
          "description": "Saw the three headed monkey!",
          "name": "Savegame",
        },
      },
      "type": "FOREMAN_SNAPSHOT_MANAGEMENT_SNAPSHOT_UPDATE_REQUEST",
    },
  ],
  Array [
    Object {
      "payload": Object {
        "key": "FOREMAN_SNAPSHOT_MANAGEMENT_SNAPSHOT_UPDATE_SUCCESS",
        "toast": Object {
          "key": "FOREMAN_SNAPSHOT_MANAGEMENT_SNAPSHOT_UPDATE_SUCCESS",
          "message": "Successfully updated Snapshot \\"Savegame\\"",
          "type": "success",
        },
      },
      "type": "toasts/addToast",
    },
  ],
  Array [
    Object {
      "payload": Object {
        "host": Object {
          "id": 42,
          "name": "deep.thought",
        },
        "id": "snapshot-0815",
      },
      "response": "some-data",
      "type": "FOREMAN_SNAPSHOT_MANAGEMENT_SNAPSHOT_UPDATE_SUCCESS",
    },
  ],
]
`;
